File: Java/Hard/CompileError/erect-the-fence_5.java
class Solution {
    public int[][] outerTrees(int[][] trees) {
        
        int n = trees.length;
        if (n <= 3) return trees;
        Arrays.sort(trees, (a, b) -> Integer.compare(a[0], b[0]));
        int[][] hull = new int[2 * n][2];
        int k = 0;
        for (int i = 0; i < n; i++) {
            while (k >= 2 && cross(hull[k - 2], hull[k - 1], trees[i]) < 0) {
                k--;
            }
            hull[k++] = trees[i];
        }
        for (int i = n - 2, t = k + 1; i >= 0; i--) {
            while (k >= t && cross(hull[k - 2], hull[k - 1], trees[i]) < 0) {
                k--;
            }
            hull[k++] = trees[i];
        }
        return Arrays.copyOf(hull, k - 1);
    }
}

File: Python3/Hard/RuntimeError/erect-the-fence_5.py
class Solution:
    def outerTrees(self, trees: List[List[int]]) -> List[List[int]]:
        
        trees.sort()
        def cross(o, a, b):
            return (a[0] - o[0]) * (b[1] - o[1]) - (a[1] - o[1]) * (b[0] - o[0])
        upper = []
        for t in trees:
            while len(upper) >= 2 and cross(upper[-2], upper[-1], t) > 0:
                upper.pop()
            upper.append(t)
        lower = []
        for t in reversed(trees):
            while len(lower) >= 2 and cross(lower[-2], lower[-1], t) > 0:
                lower.pop()
            lower.append(t)
        return list(set(upper + lower))